features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Endpoints/Discover/index.js

```diff
@@ -0,0 +1,329 @@
+            endpointSecurity: record.endpointSecurity,
+            maxTps: record.maxTps
+        };
+        const api = new API();
+        const promisedEndpoint = api.addEndpoint(endpointDefinition);
+        return promisedEndpoint.then(
+            response => {
+                const {name, id} = response.obj;
+                this.state.storedEndpoints.push(response.obj);
+                message.success("New endpoint " + name + " created successfully");
+                this.setState({
+                    actionButton: this.getUpdateButton()
+                });
+            }
+        ).catch(
+            error => {
+                console.error(error);
+                message.error("Error occurred while creating the endpoint!");
+                this.setState({
+                    endpointBeingAdded: false
+                });
+            }
+        )
+    }
+
+    handleUpdateEndpoint = () => {
+        const hideMessage = message.loading("Updating the Endpoint ...", 0);
+        let record = this.state.record;
+        let configObject = JSON.parse(record.endpointConfig);
+        let endpointName = configObject.namespace + "-" + record.name + "-"
+                            + record.type + "-" + configObject.urlType;
+        let storedEndpoint = this.state.storedEndpoints.find(el => el.name === endpointName);
+        if (storedEndpoint === null) {
+            message.error("Error while updating. Could not find the " + endpointName + " Endpoint!");
+            hideMessage();
+            return;
+        }
+        let endpointDefinition = {
+            id: storedEndpoint.id,
+            name: endpointName,
+            type: record.type,
+            endpointConfig: record.endpointConfig,
+            endpointSecurity: record.endpointSecurity,
+            maxTps: record.maxTps
+        };
+        const api = new API();
+        let promised_update = api.updateEndpoint(endpointDefinition);
+        promised_update.then(
+            response => {
+                if (response.status !== 200) {
+                    console.log("logging");
+                    console.log(response);
+                    message.error("Something went wrong while updating the " + endpointName + " Endpoint!");
+                    hideMessage();
+                    return;
+                }
+                message.success(endpointName + " Endpoint updated successfully!");
+                hideMessage()
+            }
+        );
+    }
+
+    checkIfEndpointExists = () => {
+        let record = this.state.record;
+        let endpointConfig = JSON.parse(record.endpointConfig);
+        let endpointName = endpointConfig.namespace + "-" + record.name + "-"
+                            + record.type + "-" + endpointConfig.urlType;
+        return this.state.storedEndpoints.some(el => el.name === endpointName);
+    }
+
+    componentDidMount() {
+        if(this.state.storedEndpoints != null){
```

Comment:
Formatting issue

Reply:
I'll send the correct pull request after also correcting these formatting issues